src/downloadmanager.cpp

```diff
@@ -1091,6 +1091,33 @@ void DownloadManager::openFile(int index)
   return;
 }
 
+void DownloadManager::openMetaFile(int index) {
+  if (index < 0 || index >= m_ActiveDownloads.size()) {
+    reportError("OpenMetaFile: invalid download index "+index);
```

Comment:
Apologies for not being clear. Because this is a shouldn't-happen case and the error text would make no sense to a user, I meant that it should be 1) a log instead of a message box, and 2) not translated. So I would suggest something like:
```
log::error("OpenMetaFile: invalid download index {}", index);
```